components/nimbus/src/lib.rs

```diff
@@ -888,16 +892,32 @@ pub struct NimbusTargetingHelper {
 }
 
 impl NimbusTargetingHelper {
-    fn new(context: Value, event_store: Arc<Mutex<EventStore>>) -> Self {
+    pub fn new<C: Serialize>(context: C, event_store: Arc<Mutex<EventStore>>) -> Self {
         Self {
-            context,
+            context: serde_json::to_value(context).unwrap(),
             event_store,
         }
     }
 
     pub fn eval_jexl(&self, expr: String) -> Result<bool> {
         evaluator::jexl_eval(&expr, &self.context, self.event_store.clone())
     }
+
+    pub(crate) fn put(&self, key: &str, value: bool) -> Self {
+        let context = if let Value::Object(map) = &self.context {
+            let mut map = map.clone();
+            map.insert(key.to_string(), Value::Bool(value));
+            Value::Object(map)
+        } else {
+            self.context.clone()
+        };
+
+        let event_store = self.event_store.clone();
+        Self {
+            context,
+            event_store,
+        }
+    }
```

Comment:
This goes away when we `experimenter` writes sticky targeting in terms of `active_experiments`.

Reply:
Created issue https://github.com/mozilla/experimenter/issues/8661.